pkg/serverless/trace/inferredspan/span_enrichment.go

```diff
@@ -217,10 +220,120 @@ func (inferredSpan *InferredSpan) EnrichInferredSpanWithS3Event(eventPayload eve
 	}
 }
 
+type DatadogPayload struct {
+	Type        string `json:"Type"`
+	Value       string `json:"Value"`
+	DataType    string `json:"dataType"`
+	BinaryValue string `json:"binaryValue"`
+	StringValue string `json:"stringValue"`
+}
+
+type TraceHeader struct {
```

Comment:
Are we sure we don't have this struct defined elsewhere? We should use it if so, if not this should be lowercased if we don't use it outside of this class